Automattic/jetpack
# Dashboard: Fix/admin console warnings

#### Changes proposed in this Pull Request:

* This PR fixes the following React-related console error message that displays on the Jetpack dashboard - `wp-admin/admin.php?page=jetpack#/dashboard`- with script debug enabled:

```
A future version of React will block javascript: URLs as a security precaution. Use event handlers instead if you can. If you need to generate unsafe HTML try using dangerouslySetInnerHTML instead. React was passed "javascript:void(0)".
```

* It does this by changing the link that is passed to be '#!', and in the process also change the a link to be a Button component (changed from a Link component in an earlier revision). The reasoning is explained in the (now striked out) text below, but in short since the link isn't an actual link a button worked better. This did involve a little styling as there didn't appear to be options available by default to make the button look like a regular link.
~* The Link component was used primarily because regular anchor links aren't really ideal for a link that doesn't actually go anywhere (technically a button would be better as I understand it), and even though the Link component renders as an anchor element it does provide extra features and benefits if needed later.~
~* Other options not included here:~
  ~* Another option was to change the links to button elements, which involved adding styling to style.scss. That would mean that should link styles change throughout the dashboard, the button 'links' would need to manually be re-styled to be the same as other links.~
  ~* Alternatively, using `event.preventListener` was a third option to prevent the link from going anywhere, before running our onClick function.~

#### Other information:


#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* On a Jetpack-connected site, enable script debugging in `wp-config.php`: `define( 'SCRIPT_DEBUG', true );`.
* With the browser developer console open, visit the Jetpack dashboard at `wp-admin/admin.php?page=jetpack#/dashboard`.
* You should see the console error mentioned above.
* Apply this PR and repeat - this time that specific console error should not be showing anymore.
